packages/editor/src/package/storage-format.ts

```diff
@@ -65,25 +91,47 @@ export function createEmptyDocument(
   }
 }
 
-export function migrate(stateBeforeMigration: unknown): {
+export function migrate(
+  stateBeforeMigration: unknown,
+  variant: EditorVariant
+): {
   migratedState: StorageFormat
   stateChanged: boolean
 } {
-  if (!t.type({ version: t.number }).is(stateBeforeMigration))
+  let migratedState: StorageFormat
+  let stateChanged = false
+
+  // Check if the state is the (old format)
+  if (
+    isValidDocument(stateBeforeMigration) &&
+    !StorageFormatType.is(stateBeforeMigration)
+  ) {
+    migratedState = {
+      id: uuid_v4(),
+      type: documentType,
+      variant,
+      domainOrigin: window.location.origin,
+      version: 0,
+      editorVersion: getEditorVersion(),
+      dateModified: getCurrentDatetime(),
+      document: stateBeforeMigration,
+    }
+    stateChanged = true
+  } else if (StorageFormatType.is(stateBeforeMigration)) {
+    // deep copy
+    migratedState = JSON.parse(
+      JSON.stringify(stateBeforeMigration)
+    ) as StorageFormat
+  } else {
     throw new Error(
-      `Missing property 'version: number' in state while trying to perform migrations. Got ${JSON.stringify(stateBeforeMigration)}`
+      `Unknown, invalid state format. Got ${JSON.stringify(stateBeforeMigration)}`
     )
+  }
 
-  // Only apply migrations for this version number and higher
-  const nextMigrationIndex = stateBeforeMigration.version
+  const nextMigrationIndex = migratedState.version
 
-  // Create deep copy
-  let migratedState = JSON.parse(JSON.stringify(stateBeforeMigration)) as {
-    version: number
-  }
-  let stateChanged = false
   for (let i = nextMigrationIndex; i < migrations.length; i++) {
-    migratedState = migrations[i](migratedState)
+    migratedState = migrations[i](migratedState, variant) as StorageFormat
```

Comment:
I would remove the `as StorageFormat` because it will have this runtime type only after the last migration ran not in between.